### PR TITLE
Update Castle destination docs

### PR DESCRIPTION
**Pull request by wallin**
### Proposed changes
Updating docs to match the latest technical changes
https://github.com/segmentio/analytics.js-integrations/pull/671

### Merge timing
ASAP once approved

### CODE REVIEW

File: src/connections/destinations/catalog/castle/index.md [hunk 1]
```diff
@@ -2,200 +2,87 @@
...
+2. Search for "Castle" in the Catalog, select it, and choose which of your sources to connect the destination to.
+3. Enter the "Publishable Key" into your Segment Settings UI which you can find in your property in the [Castle dashboard](https://dashboard.castle.io).
+4. Calls will now be visible in Castle dashboards in real-time.
 
-> info ""
-> If you request a Castle risk score for the "Logged in" event, you should **not** map that event to Castle's reserved `$login.succeeded`. Instead, [`authenticate`](https://docs.castle.io/docs/authentication-method) that event through Castle. See next section on _Requesting a risk score_.
+**NOTE**: Castle will only ingest Segment _client-side_ events at this point. Server-side events will be dropped and not processed.
 
-Here are two Ruby examples on how to track successful and failed login attempts (`context` and `integration` have been omitted for brevity):
+**NOTE**: Castle only supports web integrations via Segment, but we are working on mobile support so please stay tuned.
```
Comment (stayseesong) [inline src/connections/destinations/catalog/castle/index.md:18]:
```suggestion
> info ""
> Castle only supports web integrations through Segment, but is in the process of working on mobile support.
```

File: src/connections/destinations/catalog/castle/index.md [hunk 2]
```diff
@@ -2,200 +2,87 @@
...
-1. `identify`, preferably on the server-side
-1. _Optional:_ Use Castle's `authenticate` API to request a risk score
-1. _Recommended:_ Secure Mode
+## Getting Started
 
-## Tracking successful and failed logins
-A baseline integration of Castle includes tracking [successful and failed login attempts](https://docs.castle.io/docs/failed-logins). If you track these events using a Segment integration, you can use [Event Mapping](https://dashboard.castle.io/settings/events) to indicate which events correspond to Castle reserved events.
+1. From the Segment web app, click **Catalog**.
+2. Search for "Castle" in the Catalog, select it, and choose which of your sources to connect the destination to.
+3. Enter the "Publishable Key" into your Segment Settings UI which you can find in your property in the [Castle dashboard](https://dashboard.castle.io).
```
Comment (markzegarelli) [inline src/connections/destinations/catalog/castle/index.md:13]:
```suggestion
3. Enter the "Publishable Key" the Publishable Key field. Find the Publishable Key on the Castle dashboard.
```

File: src/connections/destinations/catalog/castle/index.md [hunk 3]
```diff
@@ -2,200 +2,87 @@
...
-> Server-side `track` events are dropped by Castle unless they contain the properties listed below. `identify` calls still create or update a user, but don't create a device if these properties are missing:
-> - `context.ip`. The user's IP address, i.e. not your server's internal IP
-> - `context.user_agent`, alternatively `context.headers` containing at least the `user_agent` field.
-> - `context.client_id`. The _Client ID_ forwarded by the web or mobile SDK.
 
-These properties are described in detail in the next section.
+## Page
 
-If you aren't tracking the properties above, you can still make the event appear in the user timeline by configuring it to _Force Track_ in the [Castle dashboard](https://dashboard.castle.io/settings/events). However, it does not attach to a device or contribute to the risk score.
+If you're not familiar with the Segment Specs, take a look to understand what the [`page` method](https://segment.com/docs/connections/spec/page/) does. An example call would look like:
```
Comment (stayseesong) [inline src/connections/destinations/catalog/castle/index.md:26]:
```suggestion
Take a look to understand what the [Page method](/docs/connections/spec/page/) does. An example call looks like:
```

File: src/connections/destinations/catalog/castle/index.md [hunk 4]
```diff
@@ -2,200 +2,87 @@
...
-## Tracking successful and failed logins
-A baseline integration of Castle includes tracking [successful and failed login attempts](https://docs.castle.io/docs/failed-logins). If you track these events using a Segment integration, you can use [Event Mapping](https://dashboard.castle.io/settings/events) to indicate which events correspond to Castle reserved events.
+1. From the Segment web app, click **Catalog**.
+2. Search for "Castle" in the Catalog, select it, and choose which of your sources to connect the destination to.
+3. Enter the "Publishable Key" into your Segment Settings UI which you can find in your property in the [Castle dashboard](https://dashboard.castle.io).
+4. Calls will now be visible in Castle dashboards in real-time.
 
-> info ""
-> If you request a Castle risk score for the "Logged in" event, you should **not** map that event to Castle's reserved `$login.succeeded`. Instead, [`authenticate`](https://docs.castle.io/docs/authentication-method) that event through Castle. See next section on _Requesting a risk score_.
+**NOTE**: Castle will only ingest Segment _client-side_ events at this point. Server-side events will be dropped and not processed.
```
Comment (markzegarelli) [inline src/connections/destinations/catalog/castle/index.md:16]:
```suggestion
> info ""
> Castle ingests Segment Client-side events. Server-side events are dropped and not processed.
> Castle supports Segment web integrations.
```

Add info callout, use active voice, consolidate with note below.

File: src/connections/destinations/catalog/castle/index.md [hunk 5]
```diff
@@ -2,200 +2,87 @@
...
 
-> **Note:** If you're concerned about sending `client_id` and `headers` to all of your active Segment integrations, instead include them in the `integrations.Castle` object to keep them private to your Castle integration.
-
-### The `client_id` property
-By forwarding a client identifier from the client-side to the server-side, you can link activity from the two sources to form a strong protection against attacks where this link is not present.
 
-The Castle JavaScript SDK (loaded by Analytics.js) forwards the client identifier as a browser cookie named `__cid`.
 
-The Castle [iOS](https://docs.castle.io/docs/sdk-ios) and [Android](https://docs.castle.io/docs/sdk-android) SDKs forward it as the HTTP header `X-Castle-Client-Id`. See the respective documentation pages for instructions on how to configure the header forwarding.
+`page` calls will be sent to Castle as `$page` events.
```
Comment (stayseesong) [inline src/connections/destinations/catalog/castle/index.md:34]:
```suggestion
`page` send to Castle as `$page` events.
```

File: src/connections/destinations/catalog/castle/index.md [hunk 6]
```diff
@@ -2,200 +2,87 @@
...
-String uuid = Utils.getDeviceId();
-OkHttpClient client = new OkHttpClient();
-Request request = new Request.Builder()
-  .url("https://api.yoursite.com/login")
-  .header("X-Castle-Client-Id", uuid)
-  .build();
-```
 
-> **Note**: If you have a client-less integration, for instance if you're using Castle to protect a customer-facing API, set `client_id` to `false`.
+`track` calls will be sent to Castle as a `$custom` events.
```
Comment (stayseesong) [inline src/connections/destinations/catalog/castle/index.md:47]:
```suggestion
`track` calls send to Castle as a `$custom` events.
```

File: src/connections/destinations/catalog/castle/index.md (hunk 1 above)
Comment (markzegarelli) [inline src/connections/destinations/catalog/castle/index.md:18]:
```suggestion

```

Remove this line and add to info callout above.

File: src/connections/destinations/catalog/castle/index.md [hunk 7]
```diff
@@ -2,200 +2,87 @@
...
-```
 
-There are example implementations on how to extract request headers in [PHP](https://github.com/castle/castle-php/blob/e93de1532ef28af17b8bf2bef350e6995a580085/lib/Castle/Request.php#L31), [Ruby](https://github.com/castle/castle-ruby), and [Java](https://github.com/castle/castle-java/blob/96cdc7469aa0995a836100c3dfd370b10f299e8c/src/main/java/io/castle/client/objects/UserInfoHeader.java#L148).
 
-## Identify
-When you call [`identify`](/docs/connections/spec/identify), a user will be created in Castle. The Segment special traits `email`, `username`, `name`, `createdAt`, `phone`, and `address` are mapped to Castle's reserved user traits.
+## Secure Mode
 
-Any additional traits will be stored on the Castle user model as _custom traits_.
+In order to prevent user information from being spoofed by a bad actor, it is highly recommended to send the user information as a signed JWT when Castle.js is used in production.
```
Comment (stayseesong) [inline src/connections/destinations/catalog/castle/index.md:55]:
```suggestion
In order to prevent user information from being spoofed by a bad actor, Castle recommends you to send the user information as a signed JWT when Castle.js is used in production.
```

File: src/connections/destinations/catalog/castle/index.md [hunk 8]
```diff
@@ -2,200 +2,87 @@
...
 
-> **Recommended:** Prevent `identify` from getting blocked in the client during an account takeover by calling `identify` from your server.
+From your backend code, you need to encode the user as a JWT and sign it using your Castle "API Secret". Then, when Castle receives the JWT, the integrity of the user data will be verified to ensure that the data isn't being tampered with.
```
Comment (stayseesong) [inline src/connections/destinations/catalog/castle/index.md:57]:
```suggestion
From your backend code, you need to encode the user as a JWT and sign it using your Castle **API Secret**. Then, when Castle receives the JWT, the integrity of the user data will be verified to ensure that the data isn't being tampered with.
```

File: src/connections/destinations/catalog/castle/index.md [hunk 9]
```diff
@@ -2,200 +2,87 @@
...
-def user_token(user_id)
-  OpenSSL::HMAC.hexdigest("sha256", "YOUR_CASTLE_API_SECRET", user_id.to_s)
-end
-```
-
-## Requesting a risk score
-Castle's adaptive authentication tells you whether to allow access, initiate a second factor of authentication, or log out the user.
 
-Since all Segment calls are called asynchronously, you'll need to use Castle's native SDKs to perform [adaptive authentication](https://docs.castle.io/docs/authentication-method).
+When Castle receives a JWT version of the user object, its contents will override the user object sent the standard Segment way.
```
Comment (stayseesong) [inline src/connections/destinations/catalog/castle/index.md:88]:
```suggestion
When Castle receives a JWT version of the user object, its contents override the user object sent the standard Segment way.
```

File: src/connections/destinations/catalog/castle/index.md [hunk 10]
```diff
@@ -2,200 +2,87 @@
...
-  event: 'Failed to log in',
-  properties: {
-    email: 'johan@example.com'
-  }
-)
-```
-
-> info ""
-> Segment requires either `user_id` or `anonymous_id` for the request to be processed. If you don't know which user generated the failed login create a UUID and provide it as `anonymous_id`
+***
```
Comment (markzegarelli) [inline src/connections/destinations/catalog/castle/index.md:20]:
```suggestion
```

File: src/connections/destinations/catalog/castle/index.md (hunk 3 above)
Comment (markzegarelli) [inline src/connections/destinations/catalog/castle/index.md:26]:
```suggestion
If you're not familiar with the Segment Specs, take a look to understand what the [Page call](/docs/connections/spec/page/) does. An example call looks like:
```

Relative URLs for other docs pages

File: src/connections/destinations/catalog/castle/index.md (hunk 4 above)
Comment (stayseesong) [inline src/connections/destinations/catalog/castle/index.md:16]:
```suggestion
> warning ""
> Castle only ingests Segment _client-side_ events at this point. Server-side events are dropped and won't be processed.
```

**Review by stayseesong — CHANGES_REQUESTED** [review]

File: src/connections/destinations/catalog/castle/index.md (hunk 5 above)
Comment (markzegarelli) [inline src/connections/destinations/catalog/castle/index.md:34]:
```suggestion
Segment sends Page calls to Castle as `$page` events.
```
Formatting for Segment methods / calls, and active voice.

File: src/connections/destinations/catalog/castle/index.md [hunk 11]
```diff
@@ -2,200 +2,87 @@
...
 
-```ruby
-client_id =
-  request.cookies['__cid'] ||
-  request.headers['X-Castle-Client-Id']
-```
+## Track
 
-On **iOS**, forward the device UUID as client identifier:
+If you're not familiar with the Segment Specs, take a look to understand what the [Track method](https://segment.com/docs/connections/spec/track/) does. An example call would look like:
```
Comment (markzegarelli) [inline src/connections/destinations/catalog/castle/index.md:39]:
```suggestion
If you're not familiar with the Segment Specs, take a look to understand what the [Track method](/docs/connections/spec/track/) does. An example call looks like:
```

File: src/connections/destinations/catalog/castle/index.md (hunk 6 above)
Comment (markzegarelli) [inline src/connections/destinations/catalog/castle/index.md:47]:
```suggestion
Segment sends Track calls to Castle as a `$custom` events.
```

File: src/connections/destinations/catalog/castle/index.md (hunk 7 above)
Comment (markzegarelli) [inline src/connections/destinations/catalog/castle/index.md:55]:
```suggestion
Send user information as a signed JWT when you use Castle in production. This prevents bad actors from spoofing any user information.
```
Simplify the wording here.

File: src/connections/destinations/catalog/castle/index.md (hunk 8 above)
Comment (markzegarelli) [inline src/connections/destinations/catalog/castle/index.md:57]:
```suggestion
In your backend code, encode the user as a JWT and sign it using your Castle "API Secret". When Castle receives the JWT, the integrity of the user data is verified to ensure that the data wasn't tampered with.
```

File: src/connections/destinations/catalog/castle/index.md [hunk 12]
```diff
@@ -2,200 +2,87 @@
...
-> **Note:** If you call `authenticate` to obtain a risk score, you do *not* need to call `identify` from the server-side. Instead, `authenticate` provides a way to attach `traits` in the same call.
 
-## Secure Mode
-Enable Secure Mode to prevent fraudsters from impersonating your users.
 
-> **Note:** Secure Mode is highly encouraged for production deployments, but can wait until after a completed proof a concept.
-To enable Secure Mode in Analytics.js, you pass in the `secure` variable by rendering it in your server-side templates. The `secure` field should be a SHA256 hash of your Castle API Secret and the user ID.
-
-Here's an JavaScript example of an `identify` call with Secure Mode being rendered with Ruby server-side templating language:
+You then need to transfer the `user_jwt` object to your frontend either via a separate API call, or by injecting the code using a templating language:
```
Comment (markzegarelli) [inline src/connections/destinations/catalog/castle/index.md:70]:
```suggestion
Transfer the `user_jwt` object to your frontend through a separate API call, or by injecting the code using a templating language:
```

File: src/connections/destinations/catalog/castle/index.md (hunk 9 above)
Comment (markzegarelli) [inline src/connections/destinations/catalog/castle/index.md:88]:
```suggestion
When Castle receives a JWT version of the user object, its contents override the user object sent the standard Segment way.
```

**Review by markzegarelli — CHANGES_REQUESTED** [review]
Thanks @wallin!

I've requested some changes to help this doc fit in with the rest of our docset.

File: src/connections/destinations/catalog/castle/index.md [hunk 13]
```diff
@@ -2,200 +2,87 @@
 title: Castle Destination
 id: 56a8f566e954a874ca44d3b0
 ---
-Once you enable the Castle integration, the [Castle JavaScript snippet](https://docs.castle.io/docs/sdk-browser) is placed on your website, and user data starts appearing in the Castle dashboard.
-Client-side tracking works out of the box, however **your existing server-side calls need to be extended** with data from the incoming request.
+[Castle](https://castle.io/?utm_source=segmentio&utm_medium=docs&utm_campaign=partners){:target="_blank"} monitors every step of the customer journey to help visualize and proactively block fraud that would otherwise fly under the radar. Types of fraud or abuse that can be managed include bots, fake accounts, multi-accounting, and account sharing.
 
-Castle supports calling `identify`, `page`, `screen`, and `group`. Castle does *not* support the `alias` call.
+This destination is maintained by Castle.
```
Comment (wallin) [inline src/connections/destinations/catalog/castle/index.md:7]:
```suggestion
Castle maintains this destination. For any issues with the destination, contact the [Castle support team](mailto:support@castle.io).
```

File: src/connections/destinations/catalog/castle/index.md [hunk 14]
```diff
@@ -2,200 +2,87 @@
...
-A baseline integration of Castle includes tracking [successful and failed login attempts](https://docs.castle.io/docs/failed-logins). If you track these events using a Segment integration, you can use [Event Mapping](https://dashboard.castle.io/settings/events) to indicate which events correspond to Castle reserved events.
+1. Navigate to **Connections > Catalog** in the Segment web app.
+2. Search for *Castle* in the **Destinations** tab of the catalog, and select it, and click **Configure Castle**. 
+3. Choose the sources you want to connect the destination to.
+3. Enter the "Publishable Key" the Publishable Key field. Find the Publishable Key on the Castle dashboard.
+Calls are now visible in Castle dashboards in real-time.
 
-> info ""
-> If you request a Castle risk score for the "Logged in" event, you should **not** map that event to Castle's reserved `$login.succeeded`. Instead, [`authenticate`](https://docs.castle.io/docs/authentication-method) that event through Castle. See next section on _Requesting a risk score_.
+**NOTE**: Castle will only ingest Segment _client-side_ events at this point. Server-side events will be dropped and not processed.
```
Comment (wallin) [inline src/connections/destinations/catalog/castle/index.md:17]:
```suggestion
> info ""
> Castle ingests Segment Client-side events. Server-side events are dropped and not processed.
> Castle only supports web integrations through Segment, but is in the process of working on mobile support.
```

**Comment by wallin** [comment]
Thank you for great feedback @markzegarelli @stayseesong! I've now updated the PR and hopefully addressed all your comments

**Comment by wallin** [comment]
@markzegarelli @stayseesong Thanks for merging this. How do I update the available options in the settings section here: https://segment.com/docs/connections/destinations/catalog/castle/
